dvc/repo/init.py

```diff
@@ -44,7 +44,17 @@ def _welcome_message():
     logger.info(msg)
 
 
-def init(root_dir=os.curdir, no_scm=False, force=False):
+def find_scm_dir(path):
+    from git import Repo
+    from git.exc import InvalidGitRepositoryError
+
+    try:
+        return Repo(path, search_parent_directories=True).working_tree_dir
+    except InvalidGitRepositoryError:
+        raise InitError("Could not find Git repository in parent dirs")
+
+
```

Comment:
It is really weird that you put scm-related logic in repo/init.py. Also, you call it `find_scm_dir`, but you are only looking for git repo. Maybe we could move this logic to scm/ ?